ote_sdk/ote_sdk/utils/argument_checks.py

```diff
@@ -0,0 +1,206 @@
+    if not isinstance(parameter, expected_type):
+        parameter_type = type(parameter)
+        raise ValueError(
+            f"Unexpected type of '{parameter_name}' parameter, expected: {expected_type}, actual: {parameter_type}"
+        )
+
+
+def check_required_parameters_type(parameter_name_expected_type: list):
+    """
+    Function raises ValueError exception if required parameters have unexpected type
+    :param parameter_name_expected_type: list with tuples that contain parameter, name for exception message and
+    expected type
+    """
+    for parameter, name, expected_type in parameter_name_expected_type:
+        check_parameter_type(
+            parameter=parameter, parameter_name=name, expected_type=expected_type
+        )
+
+
+def check_optional_parameters_type(parameter_name_expected_type: list):
+    """
+    Function checks if optional parameters exist and raises ValueError exception if one of them has unexpected type
+    :param parameter_name_expected_type: list with tuples that contain optional parameter, name for exception message
+    and expected type
+    """
+    for parameter, name, expected_type in parameter_name_expected_type:
+        if parameter:
+            check_parameter_type(
+                parameter=parameter, parameter_name=name, expected_type=expected_type
+            )
+
+
+def check_required_and_optional_parameters_type(
+    required_parameters: list, optional_parameters: list
+):
+    """Function raises ValueError exception if required or optional parameter has unexpected type"""
+    check_required_parameters_type(required_parameters)
+    check_optional_parameters_type(optional_parameters)
+
+
+def check_nested_elements_type(iterable, parameter_name, expected_type):
+    """Function raises ValueError exception if one of elements in collection has unexpected type"""
+    for element in iterable:
+        check_parameter_type(
+            parameter=element,
+            parameter_name=f"nested {parameter_name}",
+            expected_type=expected_type,
+        )
+
+
+def check_several_lists_elements_type(parameter_name_expected_type: list):
+    """
+    Function checks if parameters lists exist and raises ValueError exception if lists elements have unexpected type
+    :param parameter_name_expected_type: list with tuples that contain parameter with nested elements, name for
+    exception message and expected type
+    """
+    for parameter, name, expected_type in parameter_name_expected_type:
+        if parameter:
+            check_nested_elements_type(
+                iterable=parameter, parameter_name=name, expected_type=expected_type
+            )
+
+
+def check_parameter_str_class_name(parameter, parameter_name, expected_class_names):
+    """Function raises ValueError exception if string class name is not equal to expected"""
+    parameter_class_name = type(parameter).__name__
+    if (parameter_class_name in expected_class_names) is False:
```

Comment:
```suggestion
    if parameter_class_name not in expected_class_names:
```

Reply:
This function is replaced by "check_is_parameter_like_dataset"